writer.go

```diff
@@ -3,64 +3,189 @@ package log
 import (
 	"io"
 	"sync"
-	"time"
 )
 
+var MaxWriterBuffer = 16 * 1024 * 1024
```

Comment:
my worry there is that since there is really no backpressure, if a writer starts lagging a little bit, it can go through 500msgs very quickly... 

we can try 512KB and see how it goes.